src/object/srv_ec_aggregate.c

```diff
@@ -2204,8 +2247,8 @@ agg_object(daos_handle_t ih, vos_iter_entry_t *entry,
 		goto out;
 	}
 
-	rc = ds_pool_check_dtx_leader(info->api_pool, &entry->ie_oid,
-				      info->api_pool->sp_map_version, true);
+	rc = agg_obj_is_leader(info->api_pool, &oca, &entry->ie_oid,
+			       info->api_pool->sp_map_version);
```

Comment:
Right. I will fix it.

Reply:
not sure why you change this? I thought entry->ie_oid.id_shard >= oca.u.ec.e_k)  make sure this will not happen on data shard?

Reply:
EC aggregation should not depends on DTX leader which can be on data shard (via this patch), but EC leader must be one of related parity nodes.

Reply:
not a problem of this patch, but seems there is possibility that two parity servers both think the other parity server is the EC agg leader, if the pool map version is not same on them?
for example, 4P2G1 EC obj, with shards 0~5, 
on server of shard 4 with pool map ver 10 all are alive and it check shard 5 should be EC agg leader so shard 4 will do nothing;
at that time, on server of shard 5 with pool map ver 11, it checks itself failed or in-rebuilding, then it think shard 4 should be the leader so shard 5 will do nothing;
both server will not prevent the EC agg epoch move ahead but actually did not do EC agg work.
this seems also possible at original code, is it true? @wangdi1

Reply:
just FYI, per the discussion with Fanyong, a simple workable method is always only let the highest parity shard to be the obj's EC agg leader, if the last parity shard failed or in rebuilding then skip that obj's EC agg (and set ap_obj_skipped flag). Could be refined by a following patch

Reply:
>on server of shard 4 with pool map ver 10 all are alive and it check shard 5 should be EC agg leader so shard 4 will do nothing;
at that time, on server of shard 5 with pool map ver 11, it checks itself failed or in-rebuilding, then it think shard 4 should be the leader so shard 5 will do nothing;
both server will not prevent the EC agg epoch move ahead but actually did not do EC agg work.

this is just temporary stage, and their pool map version will get consistent pretty soon,  so it is fine to not do EC aggregation and not proceed EC aggregation epoch(with skipped flag), IMHO.

Reply:
"so it is fine to not do EC aggregation and not proceed EC aggregation epoch(with skipped flag)"
the problem is in that case (both parity shards both think the other parity shard is the leader) it will skip the obj's EC agg but will not set the skipped flag.